HousingRepairsOnlineApi/Gateways/AzureStorageGateway.cs

```diff
@@ -28,5 +29,40 @@ public async Task<string> UploadBlob(string base64Img, string fileExtension)
 
             return blobClient.Uri.AbsoluteUri;
         }
+
+        public async Task<string> GetServiceSasUriForBlob(string fileName, int daysUntilExpiry, string storedPolicyName = null)
+        {
+
+            var blobClient = storageContainerClient.GetBlobClient(fileName);
+
+            if (blobClient.CanGenerateSasUri)
+            {
+                var sasBuilder = new BlobSasBuilder
+                {
+                    BlobContainerName = storageContainerClient.Name,
+                    BlobName = blobClient.Name,
+                    Resource = "b"
+                };
+
+                if (storedPolicyName == null)
+                {
+                    sasBuilder.ExpiresOn = DateTimeOffset.UtcNow.AddDays(daysUntilExpiry);
+                    sasBuilder.SetPermissions(BlobSasPermissions.Read);
+                }
+                else
+                {
+                    sasBuilder.Identifier = storedPolicyName;
+                }
+
+                var sasUri = blobClient.GenerateSasUri(sasBuilder);
+                Console.WriteLine("SAS URI for blob is: {0}", sasUri);
+                return sasUri.AbsoluteUri;
+            }
+            else
+            {
+                throw new Exception(
+                    "BlobClient must be authorized with Shared Keycredentials to create a service SAS.");
```

Comment:
A space is need in `Keycredentials`
```suggestion
                    "BlobClient must be authorized with Shared Key credentials to create a service SAS.");
```